usr/share/rear/layout/prepare/GNU/Linux/130_include_filesystem_code.sh

```diff
@@ -131,17 +131,47 @@ function create_fs () {
             fi
             ;;
         (xfs)
+            Log "Begin generating code to create XFS on $device ..."
             # Cleanup disk partition:
             echo "$cleanup_command" >> "$LAYOUT_CODE"
 
             # Load xfs options from configuration files saved during
             # 'rear mkbackup/mkrescue' by xfs_info.
             # xfs info is called in 230_filesystem_layout.sh (layout/prepare)
             # xfs_opts will be used as additional parameter for mkfs.xfs and
-            # ensures that xfs filesystem will be created exactly as original.
+            # ensures that xfs filesystem will be created exactly as original
+            # unless the user has explicitly specified XFS filesystem options:
             local xfs_opts
-            xfs_opts=$(xfs_parse $LAYOUT_XFS_OPT_DIR/$(basename ${device}.xfs))
-
+            local xfs_device_basename="$( basename $device )"
+            # Only uppercase letters and digits are used to ensure mkfs_xfs_device_options_variable_name is a valid bash variable name
+            # even in case of complicated device nodes e.g. things like /dev/mapper/SIBM_2810XIV_78033E7012F-part3 
+            # cf. current_orig_device_basename_alnum_uppercase in layout/prepare/default/300_map_disks.sh
+            local xfs_device_basename_alnum_uppercase="$( echo $xfs_device_basename | tr -d -c '[:alnum:]' | tr '[:lower:]' '[:upper:]' )"
+            # cf. predefined_input_variable_name in the function UserInput in lib/_input-output-functions.sh
+            local mkfs_xfs_device_options_variable_name="MKFS_XFS_OPTIONS_$xfs_device_basename_alnum_uppercase"
+            # Set which options to use for mkfs.xfs:
+            if test "${!mkfs_xfs_device_options_variable_name:-}" ; then
+                # When the user has specified device specific options for mkfs.xfs e.g. in MKFS_XFS_OPTIONS_SDA2 use that:
+                Log "Using device specific options for mkfs.xfs as specified in $mkfs_xfs_device_options_variable_name"
```

Comment:
IMHO the log output should be more specific and explain that it is overriding the detected settings, e.g. `Overriding mkfs.xfs options from $mkfs_xfs_device_options_variable_name variable` or something like this.

Reply:
Overriding only happens when there is a file /var/lib/rear/layout/xfs/$device.xfs
so that I could enhance the Log messaging and Log different messages
depending on whether or not /var/lib/rear/layout/xfs/$device.xfs exists.